src/visualizers/VisualizerHistogram.ts

```diff
@@ -0,0 +1,211 @@
+    params = {
+        binSize: {
+            value: this.binSize,
+            forceType: 'integer',
+            displayName: 'Bin Size',
+            required: true,
+        },
+
+        firstIndex: {
+            value: this.firstIndex,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: true,
+        },
+
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'How many terms of the series',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        if (this.params.firstIndex.value < 1) {
+            status.isValid = false
+            status.errors.push('First index can not be less than 1')
+        }
+
+        return status
+    }
+
+    //find the largest value of the sequence asked for
+    largestValue(): bigint {
+        let largest_value = 0n
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            const value = this.seq.getElement(i)
+            if (i == 0) {
+                largest_value = value
+            } else if (value > largest_value) {
+                largest_value = value
+            }
+        }
+        return largest_value
+    }
+
+    //create an array with the number of factor of
+    //each element at the corresponding index of the array
+    factorArray(): number[] {
+        const factorArray = []
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [, power] of factors) {
+                    counter += Number(power)
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    //create an array with the frequency of each number
+    //of factors in the corresponding index
+    orderedFactorArray(): number[] {
+        const orderedFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            orderedFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            orderedFactorArray[factorArray[i]]++
+        }
+
+        return orderedFactorArray
+    }
+
+    //create an array tp change the bin size of the
```

Comment:
"tp" should be "to".

Reply:
Fixed in https://github.com/numberscope/frontscope/pull/183/commits/2f9c71de41cf3861c85cff5de34f4420bcea5421.